libs/vkd3d/device.c

```diff
@@ -4750,9 +4754,12 @@ static void d3d12_device_caps_init_shader_model(struct d3d12_device *device)
          * FP16, Denorm modes (float controls extension)
          */
         if (device->device_info.float16_int8_features.shaderFloat16 &&
-            device->device_info.subgroup_extended_types_features.shaderSubgroupExtendedTypes)
+                device->device_info.storage_16bit_features.storageBuffer16BitAccess &&
+                device->device_info.subgroup_extended_types_features.shaderSubgroupExtendedTypes)
```

Comment:
Do we have enough 16-bit support in dxil-spirv to enable `Native16BitShaderOpsSupported`?

Reply:
Yes, I think so.

Reply:
Guess we should write a test and enable that at some point then, but can be done separately.